### DISCUSSION TITLE
TimeToSoC seems to be missing from MQTT

### DISCUSSION THREAD

**Discussion by optio50** [Louisvdw/dbus-serialbattery#488] [Q&A] [answered]
I saw in the issues that something may have been modified in the TimeToSoC calculations. https://github.com/Louisvdw/dbus-serialbattery/pull/424
This message is no longer published to MQTT for my JBD BMS
Is this still supported?

**Comment by mr-manuel** [comment]
This should be fixed with this pull request https://github.com/Louisvdw/dbus-serialbattery/pull/468

1. Could you check, if the problem still exists, if you download a copy from my repository? Pay attention, that the config moved from `utils.py` to `config.ini`
2. What do you see, if you check with `dbus-spy`?

**Reply by optio50** [reply]
Im not sure how I should install your fork.
I was expecting to see a release archive file. In addition your fork has Louis's install script linking to his repo.

Please advise.
Thgaks for the effort you have put into this.

**Reply by mr-manuel** [reply, marked as answer]
This should work

```
cd /tmp

wget https://github.com/mr-manuel/venus-os_dbus-serialbattery/archive/refs/heads/master.zip
unzip master.zip

cp -rf /tmp/venus-os_dbus-serialbattery-master/etc/dbus-serialbattery/ /data/etc

chmod +x /data/etc/dbus-serialbattery/*.sh
chmod +x /data/etc/dbus-serialbattery/*.py
chmod +x /data/etc/dbus-serialbattery/service/run
chmod +x /data/etc/dbus-serialbattery/service/log/run

bash /data/etc/dbus-serialbattery/reinstalllocal.sh
```

**Reply by optio50** [reply]
The TimeToSoC are published and working. Thank you.
However, BATTERY_CELL_DATA_FORMAT does not seem to be respected.
I have tried both 1 & 3, but only 2 is published regardless of format selection.

; Select the format of cell data presented on dbus [Valid values 0,1,2,3]
; 0 Do not publish all the cells (only the min/max cell data as used by the default GX)
; 1 Format: /Voltages/Cell; (also available for display on Remote Console)
; 2 Format: /Cell/#/Volts
; 3 Both formats 1 and 2
BATTERY_CELL_DATA_FORMAT = 1

Edit: TimeToSoC
Still testing but TimeToSoC only shows for discharge.

Edit #2:

Nevermind. All seems to be working.
Sorry :-)

Edit #3
dang, something is weird with TimeToSoC
it still doesn't show while charging and the old discharge values get stuck in dbus.
I can try reinstalling louis's 14.2 and check it out

**Reply by mr-manuel** [reply]
Have you set this to true?

```
; Include TimeToSoC points when moving away from the SoC point [Valid values True, False]
; These will be as negative time. Disabling this improves performance slightly
TIME_TO_SOC_INC_FROM = True
```
